lib/ansible/plugins/callback/default.py

```diff
@@ -44,8 +46,31 @@ def __init__(self):
         self._last_task_banner = None
         self._last_task_name = None
         self._task_type_cache = {}
+        self._display_skipped_hosts = None
         super(CallbackModule, self).__init__()
 
+    def display_skipped_hosts(self):
+        """
+        Evaluates and returns dynamic configuration of display_skipped_hosts in a playbook
+        """
+        prereqs = CLI._play_prereqs()
+        self._loader = prereqs[0]
+        self._variable_manager = prereqs[2]
+        vars = self._variable_manager.get_vars(play=self._play)
+        if "display_skipped_hosts" not in vars:
+            return self.get_option("display_skipped_hosts")
+        templar = Templar(loader=self._loader, variables=vars)
+
+        display_skipped_hosts_val = templar.template(vars["display_skipped_hosts"], convert_data=True)
```

Comment:
this is not how configuration should work, the variable should be added to the callback option definition. Also see #84661 for the scope expected for variables in a callback.

Reply:
Got it. Since there's another PR open for this issue I'll close this for now